src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable jsx-a11y/no-autofocus */
+import { Dispatch, SetStateAction, useState } from 'react';
+import cn from 'classnames';
+import { Todo as TodoType } from '../../types/Todo';
+
+interface Props {
+  todo: TodoType,
+  onDelete?: (id: number) => void,
+  onUpdate?: (updateteTodo: TodoType) => void;
+  isEditing?: boolean,
+  setEditing?: Dispatch<SetStateAction<number | null>>,
+
+}
+
+export const Todo: React.FC<Props> = (
+  {
+    todo,
+    onDelete,
+    onUpdate,
+    isEditing,
+    setEditing,
+  },
+) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [title, setTitle] = useState(todo.title);
+
+  const handleDelete = async () => {
+    setIsLoading(true);
+
+    try {
+      await onDelete?.(todo.id);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const toggleComplete = async () => {
+    setIsLoading(true);
+    try {
+      await onUpdate?.({ ...todo, completed: !todo.completed });
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const saveUpdatedTitle = async () => {
+    setIsLoading(true);
+
+    try {
+      await onUpdate?.({ ...todo, title });
+    } finally {
+      setEditing?.(null);
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={cn('todo',
+        {
+          'todo completed': todo.completed,
+        })}
+      onDoubleClick={() => setEditing?.(todo.id)}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={toggleComplete}
+        />
+      </label>
+
+      {isEditing
+        ? (
+          <form
+            onSubmit={e => {
```

Comment:
In my opinion, it's better to take this to the separate handler

Reply:
> In my opinion, it's better to take this to the separate handler

Good point